grpc/private/gateway.go

```diff
@@ -0,0 +1,83 @@
+package private
+
+import (
+	"bytes"
+	"net"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/keratin/authn-server/app"
+	authnpb "github.com/keratin/authn-server/grpc"
+	"github.com/keratin/authn-server/grpc/internal/gateway"
+	"github.com/keratin/authn-server/grpc/public"
+	"github.com/keratin/authn-server/server/views"
+	"golang.org/x/net/context"
+	grpc "google.golang.org/grpc"
+)
+
+func RunPrivateGateway(ctx context.Context, app *app.App, r *mux.Router, conn *grpc.ClientConn, l net.Listener) error {
+
+	gmux := runtime.NewServeMux(
+		runtime.WithForwardResponseOption(gateway.CookieSetter(app.Config)), // Cookies always have to go first
+		runtime.WithForwardResponseOption(gateway.StatusCodeMutator),
+		// Workaround this limitation: https://github.com/grpc-ecosystem/grpc-gateway/issues/920.
+		// Go's JSON encoder doesn't convert (u)int64 to strings silently.
+		runtime.WithMarshalerOption(runtime.MIMEWildcard, gateway.JSONMarshaler()),
+		runtime.WithMetadata(gateway.CookieAnnotator(app)),
+	)
+
+	r.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path == "/" {
```

Comment:
this looks like the only difference between here and `handlers.GetRoot`. why is it necessary? is it relevant to both locations?

Reply:
My bad. `handlers.GetRoot` can definitely be used.

Reply:
There's something awkward with the root path due to an idiosyncrasy of Gorilla's `PathPrefix`. Whenever `PathPrefix` is used, StrictSlash is automatically set to `true`. So if the `AUTHN_URL` is `/authn`, then it can only be reached on `/authn/` and never `/authn`. FWIW, this is the current behavior on `master` anyways.